pkg/apiserver/api/v1/account.proto

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2019 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+package kubecarrier.api.v1;
+option go_package = "v1";
+
+import "google/api/annotations.proto";
+import "google/protobuf/timestamp.proto";
+
+import "meta.proto";
+import "types.proto";
+
+message Account {
+  ObjectMeta metadata = 1;
+  AccountSpec spec = 2;
+  AccountStatus status = 3;
+}
+
+message AccountSpec {
+  AccountMetadata metadata = 1;
+  repeated string roles = 2;
+  repeated Subject subjects = 3;
+}
+
+message AccountMetadata {
+  string displayName = 1;
+  string description = 2;
+  string shortDescription = 3;
+  Image logo = 4;
+  Image icon = 5;
+}
+
+message Subject {
+  string kind = 1;
+  string apiGroup = 2;
+  string name = 3;
+  string namespace = 4;
+}
+
+message AccountStatus {
```

Comment:
Is this too much detail exposed in the API server? Why would the end-user care which namespace the account is in? Or the accounts `observedGeneration`?

Reply:
For namespace, user cares, because user should know where all his/her objects/ service instances will be created.
For `observedGeneration`, WDYT @thetechnick @aborilov ?

Reply:
Users don't need to know the namespace, as we don't expose access to "Namespaces" via the KubeCarrier API.
While not critical, please remove the field.

Reply:
fixed

Reply:
Why are we exposing `phase` here which is deprecated in the Kubernetes?

Reply:
hmm, you are right, removed